src/coreclr/src/debug/daccess/dacdbiimpl.cpp

```diff
@@ -3570,6 +3571,137 @@ HRESULT DacDbiInterfaceImpl::GetDelegateTargetObject(
+    for (uint32_t i = 0; i < (uint32_t)heapsToEnumerate.Size(); i++)
+    {
+        LOG((LF_CORDB, LL_INFO10000, "DDBII::EMRFLA: LoaderHeap 0x%x\n", heapsToEnumerate[i].GetAddr()));
+        heapsToEnumerate[i]->EnumPageRegions(TrackMemoryRangeHelper, rangeAccumAsTaddr);
+    }
+}
+
+void DacDbiInterfaceImpl::EnumerateMemRangesForJitCodeHeaps(CQuickArrayList<COR_MEMORY_RANGE> *rangeAcummulator)
+{
+    // The JIT Code heap will always have
+    EEJitManager *pEM = ExecutionManager::GetEEJitManager();
+    _ASSERT(pEM != NULL && pEM->m_pCodeHeap.IsValid());
+
+    PTR_HeapList pHeapList = pEM->m_pCodeHeap;
+    while (pHeapList != NULL)
+    {
+        CodeHeap *pHeap = pHeapList->pHeap;
+        DacpJitCodeHeapInfo jitCodeHeapInfo = DACGetHeapInfoForCodeHeap(pHeap);
+
+        switch (jitCodeHeapInfo.codeHeapType)
+        {
+            case CODEHEAP_LOADER:
+            {
+                TADDR targetLoaderHeap = CLRDATA_ADDRESS_TO_TADDR(jitCodeHeapInfo.LoaderHeap);
+                LOG((LF_CORDB, LL_INFO10000,
+                    "DDBII::EMRFJCH: LoaderCodeHeap 0x%x with LoaderHeap at 0x%x\n",
+                    PTR_HOST_TO_TADDR(pHeap), targetLoaderHeap));
+                PTR_ExplicitControlLoaderHeap pLoaderHeap = PTR_ExplicitControlLoaderHeap(targetLoaderHeap);
+                pLoaderHeap->EnumPageRegions(TrackMemoryRangeHelper, TO_TADDR(rangeAcummulator));
+                break;
+            }
+
+            case CODEHEAP_HOST:
+            {
+                LOG((LF_CORDB, LL_INFO10000,
+                    "DDBII::EMRFJCH: HostCodeHeap 0x%x\n",
+                    PTR_HOST_TO_TADDR(pHeap)));
+                rangeAcummulator->Push({
+                    CLRDATA_ADDRESS_TO_TADDR(jitCodeHeapInfo.HostData.baseAddr),
+                    CLRDATA_ADDRESS_TO_TADDR(jitCodeHeapInfo.HostData.currentAddr)
+                });
+                break;
+            }
+
+            default:
+            {
+                LOG((LF_CORDB, LL_INFO10000, "DDBII::EMRFJCH: unknown heap type at 0x%x\n\n", pHeap));
+                break;
+            }
+        }
+
+        pHeapList = pHeapList->GetNext();
+    }
+}
+
+HRESULT DacDbiInterfaceImpl::GetLoaderHeapMemoryRanges(DacDbiArrayList<COR_MEMORY_RANGE> *pRanges)
+{
+    LOG((LF_CORDB, LL_INFO10000, "DDBII::GLHMR\n"));
+    DD_ENTER_MAY_THROW;
+
+    HRESULT hr = S_OK;
+
+    EX_TRY
+    {
+        CQuickArrayList<COR_MEMORY_RANGE> memoryRanges;
+
+        // Anything that's loaded in the SystemDomain or into the main AppDomain's default context in .NET Core
+        // and after uses only one global allocator. Enumerating that one is enough for most purposes.
+        PTR_LoaderAllocator pGlobalAllocator = SystemDomain::System()->GetLoaderAllocator();
+        _ASSERTE(pGlobalAllocator);
+        EnumerateMemRangesForLoaderAllocator(pGlobalAllocator, &memoryRanges);
+        // TODO: This doesn't consider any uses of AssemblyLoadingContexts (Unloadable or not). Each context has
```

Comment:
Nit: I recommend leaving out the "TODO", the rest of the comment is fine. "TODO" in a comment usually winds up meaning the opposite of its naïve interpretation - it is the work you don't plan to do ; ) Same thing for "TODO"s below.